lib/phonebook/ui/pages/association_editor_page/member_editable_card.dart

```diff
@@ -16,24 +20,32 @@ import 'package:myecl/phonebook/tools/constants.dart';
 import 'package:qlevar_router/qlevar_router.dart';
 
 class MemberEditableCard extends HookConsumerWidget {
-  const MemberEditableCard({super.key, required this.member});
+  const MemberEditableCard(
+      {super.key, required this.member, required this.association});
 
   final CompleteMember member;
+  final Association association;
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final profilePicture = ref.watch(profilePictureProvider);
-    final association = ref.watch(associationProvider);
     final associationNotifier = ref.watch(asyncAssociationProvider.notifier);
     final associationMembersNotifier =
         ref.watch(associationMemberListProvider.notifier);
     final roleTagsNotifier = ref.watch(rolesTagsProvider.notifier);
-    final isEditNotifier = ref.watch(isEditProvider.notifier);
+    final membershipNotifier = ref.watch(membershipProvider.notifier);
     final completeMemberNotifier = ref.watch(completeMemberProvider.notifier);
     void displayToastWithContext(TypeMsg type, String msg) {
       displayToast(context, type, msg);
     }
 
+    Membership? assoMembership = member.memberships.firstWhereOrNull(
+        (memberships) =>
+            memberships.associationId == association.id &&
+            memberships.mandateYear == association.mandateYear);
+
+    assoMembership ??= Membership.empty();
```

Comment:
You can use .firstWhere, and pass an orElse argument : https://api.flutter.dev/flutter/dart-core/Iterable/firstWhere.html

Reply:
[fixed](https://github.com/aeecleclair/Titan/pull/41/commits/6af907212309ca853ed45e9ed17d1192ea35eac1)